cmd/nerdctl/network_ls.go

```diff
@@ -65,86 +50,14 @@ func networkLsAction(cmd *cobra.Command, args []string) error {
 	if err != nil {
 		return err
 	}
-	w := cmd.OutOrStdout()
-	var tmpl *template.Template
 	format, err := cmd.Flags().GetString("format")
 	if err != nil {
 		return err
 	}
-	switch format {
-	case "", "table", "wide":
-		w = tabwriter.NewWriter(cmd.OutOrStdout(), 4, 8, 4, ' ', 0)
-		if !quiet {
-			fmt.Fprintln(w, "NETWORK ID\tNAME\tFILE")
-		}
-	case "raw":
-		return errors.New("unsupported format: \"raw\"")
-	default:
-		if quiet {
-			return errors.New("format and quiet must not be specified together")
-		}
-		var err error
-		tmpl, err = formatter.ParseTemplate(format)
-		if err != nil {
-			return err
-		}
-	}
-
-	e, err := netutil.NewCNIEnv(globalOptions.CNIPath, globalOptions.CNINetConfPath)
-	if err != nil {
-		return err
-	}
-	netConfigs, err := e.NetworkList()
-	if err != nil {
-		return err
-	}
-	pp := make([]networkPrintable, len(netConfigs))
-	for i, n := range netConfigs {
-		p := networkPrintable{
-			Name: n.Name,
-			file: n.File,
-		}
-		if n.NerdctlID != nil {
-			p.ID = *n.NerdctlID
-			if len(p.ID) > 12 {
-				p.ID = p.ID[:12]
-			}
-		}
-		if n.NerdctlLabels != nil {
-			p.Labels = formatter.FormatLabels(*n.NerdctlLabels)
-		}
-		pp[i] = p
-	}
-
-	// append pseudo networks
-	pp = append(pp, []networkPrintable{
-		{
-			Name: "host",
-		},
-		{
-			Name: "none",
-		},
-	}...)
-
-	for _, p := range pp {
-		if tmpl != nil {
-			var b bytes.Buffer
-			if err := tmpl.Execute(&b, p); err != nil {
-				return err
-			}
-			if _, err = fmt.Fprintf(w, b.String()+"\n"); err != nil {
-				return err
-			}
-		} else if quiet {
-			if p.ID != "" {
-				fmt.Fprintln(w, p.ID)
-			}
-		} else {
-			fmt.Fprintf(w, "%s\t%s\t%s\n", p.ID, p.Name, p.file)
-		}
-	}
-	if f, ok := w.(formatter.Flusher); ok {
-		return f.Flush()
+	options := &types.NetworkListCommandOptions{
+		GOptions: globalOptions,
+		Quiet:    quiet,
+		Format:   format,
 	}
-	return nil
+	return network.List(options, cmd.OutOrStdout())
```

Comment:
we probably should add a `context.Context` argument and pass in `cmd.Context()`, similar to

https://github.com/containerd/nerdctl/blob/cfe96a9d17b293463ad754ebdae6f0b933d7ae39/cmd/nerdctl/images.go#L133

Reply:
done